### PR TITLE
fix rotation of callout anchor

### LINKED ISSUE #1503: Callout: anchor wrong when rotating with map (again!)
This is sort of a reincarnation of #1151, with exactly the same phenomenology ...

When using a `Pin` with `Callout`, both of which have `RotateWithMap=true`, the callout anchor is placed incorrectly when the map is rotated off the default orientation, observable e.g. in the "Add Pin Sample" (PinSample.cs).

This has been working with 3.0.2 (was fixed in #1179), but seems to be broken again now in 4.0.0-beta.1.

### PR DESCRIPTION
**Pull request by janusw**
* fixes #1503
* and at the same time cleans up the code a bit
  - use an MPoint for the symbol offset, so that we can use its rotate method
  - add an conversion method Offset.ToPoint
  - add a variant of MPoint.Rotate that otates around the origin

### CODE REVIEW

File: Mapsui.Rendering.Skia/SkiaStyles/CalloutStyleRenderer.cs [hunk 1]
```diff
@@ -31,26 +31,28 @@ public class CalloutStyleRenderer : SymbolStyle
...
-            if (viewport.Rotation != 0 && calloutStyle.SymbolOffsetRotatesWithMap)
+            if (viewport.Rotation != 0)
             {
-                var mapRotation = viewport.Rotation / 180.0 * Math.PI;
-                symbolOffsetX = (float)(Math.Cos(mapRotation) * symbolOffsetX - Math.Sin(mapRotation) * symbolOffsetY);
-                symbolOffsetY = (float)(Math.Sin(mapRotation) * symbolOffsetX + Math.Cos(mapRotation) * symbolOffsetY);
+                if (calloutStyle.RotateWithMap)
+                    rotation += (float)viewport.Rotation;
+                if (calloutStyle.SymbolOffsetRotatesWithMap)
+                    symbolOffset = symbolOffset.Rotate(-viewport.Rotation);
```
Comment (inforithmics) [inline Mapsui.Rendering.Skia/SkiaStyles/CalloutStyleRenderer.cs:48]: Now its more readable to find see what happens

File: Mapsui.Rendering.Skia/SkiaStyles/CalloutStyleRenderer.cs [hunk 2]
```diff
@@ -31,26 +31,28 @@ public class CalloutStyleRenderer : SymbolStyle
             var picture = (SKPicture)BitmapRegistry.Instance.Get(calloutStyle.BitmapId);
 
             // Calc offset (relative or absolute)
-            var symbolOffsetX = calloutStyle.SymbolOffset.IsRelative ? picture.CullRect.Width * (float)calloutStyle.SymbolOffset.X : (float)calloutStyle.SymbolOffset.X;
-            var symbolOffsetY = calloutStyle.SymbolOffset.IsRelative ? picture.CullRect.Height * (float)calloutStyle.SymbolOffset.Y : (float)calloutStyle.SymbolOffset.Y;
+            MPoint symbolOffset = calloutStyle.SymbolOffset.ToPoint();
+            if (calloutStyle.SymbolOffset.IsRelative)
+            {
+                symbolOffset.X *= picture.CullRect.Width;
```
Comment (inforithmics) [inline Mapsui.Rendering.Skia/SkiaStyles/CalloutStyleRenderer.cs:37]: The Offsets are assigned and wouldn't be wrongly optimized.

File: Mapsui.Rendering.Skia/SkiaStyles/CalloutStyleRenderer.cs [hunk 3]
```diff
@@ -31,26 +31,28 @@ public class CalloutStyleRenderer : SymbolStyle
             var picture = (SKPicture)BitmapRegistry.Instance.Get(calloutStyle.BitmapId);
 
             // Calc offset (relative or absolute)
-            var symbolOffsetX = calloutStyle.SymbolOffset.IsRelative ? picture.CullRect.Width * (float)calloutStyle.SymbolOffset.X : (float)calloutStyle.SymbolOffset.X;
-            var symbolOffsetY = calloutStyle.SymbolOffset.IsRelative ? picture.CullRect.Height * (float)calloutStyle.SymbolOffset.Y : (float)calloutStyle.SymbolOffset.Y;
+            MPoint symbolOffset = calloutStyle.SymbolOffset.ToPoint();
```
Comment (inforithmics) [inline Mapsui.Rendering.Skia/SkiaStyles/CalloutStyleRenderer.cs:34]: Now it is a point and easier to handle

**Review by inforithmics — APPROVED** [review]

**Comment by janusw** [comment]
Thanks for the review, @inforithmics!

**Pull request merged**